### PR TITLE
API Generator: support GraphQLJSONObject input for fields that are not a InputType class

### PR DESCRIPTION
**Pull request by nsams**

### CODE REVIEW

File: packages/api/cms-api/src/generator/generate-crud-input.ts [hunk 1]
```diff
@@ -280,21 +280,27 @@ export async function generateCrudInput(
...
-                } else {
+                } else if (tsType.getArrayElementTypeOrThrow().isClass()) {
                     const nestedClassName = tsType.getArrayElementTypeOrThrow().getText(tsProp);
                     const importPath = findInputClassImportPath(nestedClassName, generatorOptions, metadata);
                     imports.push({ name: nestedClassName, importPath });
                     decorators.push(`@ValidateNested()`);
                     decorators.push(`@Type(() => ${nestedClassName})`);
                     decorators.push(`@Field(() => [${nestedClassName}], ${fieldOptions})`);
+                } else {
+                    decorators.push(`@Field(() => [GraphQLJSONObject], ${fieldOptions}) //Warning: this input is not validated properly`);
```
Comment (johnnyomair) [inline packages/api/cms-api/src/generator/generate-crud-input.ts:291]:
```suggestion
                    decorators.push(`@Field(() => [GraphQLJSONObject], ${fieldOptions}) // Warning: this input is not validated properly`);
```

File: packages/api/cms-api/src/generator/generate-crud-input.ts [hunk 2]
```diff
@@ -280,21 +280,27 @@ export async function generateCrudInput(
...
+            } else if (tsType.isClass()) {
                 const nestedClassName = tsType.getText(tsProp);
                 const importPath = findInputClassImportPath(nestedClassName, generatorOptions, metadata);
                 imports.push({ name: nestedClassName, importPath });
                 decorators.push(`@ValidateNested()`);
                 decorators.push(`@Type(() => ${nestedClassName})`);
                 decorators.push(`@Field(() => ${nestedClassName}${prop.nullable ? ", { nullable: true }" : ""})`);
+            } else {
+                decorators.push(
+                    `@Field(() => GraphQLJSONObject${prop.nullable ? ", { nullable: true }" : ""}) //Warning: this input is not validated properly`,
```
Comment (johnnyomair) [inline packages/api/cms-api/src/generator/generate-crud-input.ts:302]:
```suggestion
                    `@Field(() => GraphQLJSONObject${prop.nullable ? ", { nullable: true }" : ""}) // Warning: this input is not validated properly`,
```

**Review by johnnyomair — COMMENTED** [review]
Is this warning really helpful?

**Comment by johnnyomair** [comment]
> Don't know, I just don't wanted to accept this silently

Should the warning be in the GraphQL schema (using a comment) instead? I doubt developers will review the generated code _that_ carefully...

**Comment by nsams** [comment]
> Should the warning be in the GraphQL schema (using a comment) instead? I doubt developers will review the generated code that carefully...

imho no, it's not important for the user of this api.

right, this small comment alone will not change a lot - developers using JSON in graphql need to be aware of the consequences :)

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**